ext/cartesiangrid.jl

```diff
@@ -84,7 +84,11 @@ function vizgrid2D!(plot)
     nd = embeddim($grid)
     or = coordinates(minimum($grid))
     sp = spacing($grid)
+    
     sz = size($grid)
+    if *(sz...) != length($colorant)
+      sz = sz .+ 1
+    end
```

Comment:
This fix should be implemented in the main Makie.plot! method so that it works with 2D and 3D grids.